src/libs/elektra/keyset.c

```diff
@@ -473,21 +485,21 @@ int ksDel (KeySet * ks)
 		elektraFree (ks);
 	}
 
-	return rc;
+	return 0;
 }
 
 /**
- * @internal
- *
- * KeySet object cleaner.
+ * @brief Empties a KeySet.
  *
- * Will keyDel() all contained keys, reset internal pointers and counters.
+ * This function
+ * - **does not** check or modify the reference count of `ks`
+ * - decrements the reference count of all keys contained in `ks`
+ * - deletes all keys that where only referenced by `ks`
+ * - resets size of `ks` to 0
  *
- * After this call you can use the empty keyset again.
+ * @param ks the KeySet to clear
  *
- * @param ks the keyset object to work with
- * @see ksAppendKey() for details on how keys are inserted in KeySets
- * @retval 0 on success
+ * @retval  0 on success
  * @retval -1 on failure (memory) or ks == NULL
  */
 int ksClear (KeySet * ks)
```

Comment:
Should we make `ksClear` private, or at least check the reference count if we keep it public?

Reply:
Keeping `ksClear` is fine IMO. It serves a useful purpose, namely removing all keys from a KeySet. Essentially it is a much more efficient version of:
```c
for (elektraCursor it = 0; it < ksGetSize (ks); ++it)
{
    // doesn't exist right now
    // ksRemoveAt (ks, it);

    // the current API allows only
    keyDel (ksLookup (ks, ksAtCursor (ks, it), KDB_O_POP));
}
```

Checking the reference counter would be weird, since it just modifies the contents of the KeySet. `ksAppendKey` and `ksLookup (..., KDB_O_POP)` are allowed even, if there are multiple references, so why not `ksClear`?